File: src/az/pattern/AZBurstShoot.java
package az.pattern;

import mindustry.entities.pattern.ShootPattern;
import arc.math.Rand;

public class AZBurstShoot extends ShootPattern {
    public int min;
    public int max;

    public AZBurstShoot(int min, int max, float shotDelay) {
        this.min = min;
        this.max = max;
        this.shotDelay = shotDelay;
    }

    @Override
    public void shoot(int totalShots, ShootPattern.BulletHandler handler) {
        int randNum = new Rand().random(min, max);

        for(int i = 0; i <= randNum; ++i) {
            //TODO fix the delay (MAYBE)
            handler.shoot(0, 0, 0, firstShotDelay + shotDelay * i);
        }
    }
}

File: src/az/pattern/AZBurstShoot.kt
//slish add kotlin support pls 🥺
package az.pattern

import arc.math.Rand
import mindustry.entities.pattern.ShootPattern

open class AZBurstShoot(val min: Int, val max: Int, var shotDelay: Float) : ShootPattern() {
    override fun shoot(totalShots: Int, handler: BulletHandler?) {
        val randNum = Rand().random(min, max)
        for (i in 1..randNum) {
            handler!!.shoot(0F, 0F, 0F, firstShotDelay + this.shotDelay * i)
        }
    }
}